src/ir/iteration.h

```diff
@@ -84,10 +84,13 @@ class ChildIterator {
 
 // Returns true if the current expression contains a certain kind of expression,
 // within the given depth of BFS. If depth is -1, this searches all children.
-template<typename T> bool containsChild(Expression* parent, int depth = -1) {
+template<typename T> bool containsChild(Expression* curr, int depth = -1) {
```

Comment:
Unrelated to this PR, but it looks like if `depth` is -1 this function will search no children, not all children 😕

Reply:
Oh, good catch, thanks. I deleted the option of searching all children with depth -1 because we can do the same thing with `FindAll`.